lmcache/v1/storage_backend/gds_backend.py

```diff
@@ -89,28 +94,46 @@ def get_fstype(path):
     return best_fstype
 
 
-def pack_metadata(shape, dtype, size) -> bytes:
-    # TODO: we can easily become SafeTensors compatible by taking some
-    # code from: https://github.com/vast-data/VUA/blob/main/src/vua/serdes.py,
-    # there's no need to invent a new format.
-    metadata_desc = "<QQQQ" + len(shape) * "Q"
-    if struct.calcsize(metadata_desc) > _METADATA_MAX_SIZE:
-        raise ValueError(
-            f"Metadata size {struct.calcsize(metadata_desc)} "
-            f"exceeds max size {_METADATA_MAX_SIZE}"
-        )
-    return struct.pack(
-        metadata_desc, _METADATA_VERSION, dtype_to_idx[dtype], size, len(shape), *shape
-    )
+def pack_metadata(tensor, **extra_metadata) -> bytes:
+    if tensor.dtype not in torch_dtypes:
+        raise RuntimeError(f"unhandled dtype {tensor.dtype}")
+
+    # Metadata
+    data_size = tensor.numel() * tensor.element_size()
+    tensor_meta = {
+        "dtype": torch_dtypes[tensor.dtype],
+        "shape": list(tensor.size()),
+        "data_offsets": [0, data_size],
+        "__metadata__": extra_metadata,
+    }
+    meta = {"kvcache": tensor_meta}
+    str_meta = json.dumps(meta).encode("utf-8")
+    meta_len = len(str_meta)
+
+    # Align to _METADATA_MAX_SIZE - 8
+    str_meta += b" " * (_METADATA_MAX_SIZE - 8 - meta_len)
+
+    # Pack it all up so it is sized _METADATA_MAX_SIZE exactly.
+    return struct.pack("L", len(str_meta)) + str_meta
```

Comment:
Please fix the byte-order ("<L", not "L"). I know little-endian kinda won, but still seems to be cleaner to specify it explicitly.

Reply:
Also, according to safetensors doc it must be "<Q" (64 bits unsigned)

Reply:
Fixed